components/application-role-mgt/org.wso2.carbon.identity.application.role.mgt/src/main/java/org/wso2/carbon/identity/application/role/mgt/dao/ApplicationRoleMgtDAO.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.role.mgt.dao;
+
+import org.wso2.carbon.identity.application.common.model.IdentityProvider;
+import org.wso2.carbon.identity.application.role.mgt.exceptions.ApplicationRoleManagementException;
+import org.wso2.carbon.identity.application.role.mgt.exceptions.ApplicationRoleManagementServerException;
+import org.wso2.carbon.identity.application.role.mgt.model.ApplicationRole;
+
+import java.util.List;
+
+/**
+ * Application role DAO interface.
+ */
+public interface ApplicationRoleMgtDAO {
+
+    ApplicationRole addApplicationRole(ApplicationRole applicationRole, String tenantDomain)
```

Comment:
Shall we add the method comments to keep the consistancy?

Reply:
fixed